tests/other/utils.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Other test utils."""
+
+import json
+import typing
+
+import cryptography.hazmat.primitives.asymmetric.ed25519
+import cryptography.hazmat.primitives.serialization
+import cryptography.x509
+import ops.testing
+
+from tests.utils import TEST_CA_CERT, TEST_CA_KEY, get_csr_common_name, sign_csr
+
+__all__ = ["TEST_CA_KEY", "TEST_CA_CERT", "sign_csr", "get_csr_common_name", "TlsProvider"]
+
+
+class TlsProvider:
```

Comment:
Should we rename this as `class TlsProviderFake`? So that we know it is a fake implementation for testing :D

Reply:
Indeed, I change the name to `TlsProviderStub`, thanks!